### PR TITLE
Bug 1306578 - Fix retry on pulse job ingestion

### PR DESCRIPTION
**Pull request by camd**
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1892)

### CODE REVIEW

File: tests/etl/test_job_loader.py [hunk 1]
```diff
@@ -79,6 +80,22 @@ def test_ingest_pulse_jobs(pulse_jobs, test_project, jm, result_set_stored,
     assert JobDetail.objects.count() == 2
 
 
+def test_ingest_pulse_jobs_bad_project(pulse_jobs, test_project, jm, result_set_stored,
```
Comment (edmorley) [inline tests/etl/test_job_loader.py]: Could you move this to a separate commit? :-)

File: treeherder/etl/job_loader.py [hunk 2]
```diff
@@ -58,7 +60,10 @@ def process_job_list(self, all_jobs_list):
                             logger.warn("Skipping job due to bad attribute",
                                         exc_info=1)
 
-                jobs_model.store_job_data(storeable_job_list)
+                try:
+                    jobs_model.store_job_data(storeable_job_list)
+                except(DatasetNotFoundError):
```
Comment (edmorley) [inline treeherder/etl/job_loader.py]: s/except(DatasetNotFoundError)/except DatasetNotFoundError/

I'm also surprised the `clean_revision()` call above doesn't raise `DatasetNotFoundError` too?

File: tests/etl/test_job_loader.py (hunk 1 above)
Comment (camd) [inline tests/etl/test_job_loader.py]: done

**Comment by edmorley** [comment]
> It doesn't have to be in this PR, but could we also improve the exception message? (eg add the repo name)

Could you file a bug for this? :-)
